roles/container-engine/containerd/templates/hosts.toml.j2

```diff
@@ -4,4 +4,22 @@ server = "{{ item.server | default("https://" + item.prefix) }}"
   capabilities = ["{{ ([ mirror.capabilities ] | flatten ) | join('","') }}"]
   skip_verify = {{ mirror.skip_verify | default('false') | string | lower }}
   override_path = {{ mirror.override_path | default('false') | string | lower }}
+{% if mirror.ca is defined %}
```

Comment:
I would move this logic to a task and with `set_fact` to keep the template clean. 

Also we could always provide a CA/client as list:  ca: [one], and not deal with type checking.